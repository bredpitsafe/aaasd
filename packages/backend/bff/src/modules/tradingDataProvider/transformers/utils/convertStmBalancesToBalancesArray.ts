import type { TStmBalance as TStmBalanceSDK } from '@grpc-schemas/trading_data_provider-api-sdk/services/trading_data_provider/v1/stm_api.js';
import { omit } from 'lodash-es';

import type { TStmBalance } from '../../schemas/SubscribeToStmBalances.schema.ts';

export const convertStmBalancesToBalancesArray = (balances?: TStmBalanceSDK[]): TStmBalance[] =>
    balances?.map((update) => omit(update, 'position')) ?? [];
