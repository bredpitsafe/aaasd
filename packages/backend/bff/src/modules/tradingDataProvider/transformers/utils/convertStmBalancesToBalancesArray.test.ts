import { describe, expect, it } from 'vitest';

import { stmBalance } from '../fixtures/stmBalance.ts';
import { convertStmBalancesToBalancesArray } from './convertStmBalancesToBalancesArray.ts';

describe('convertStmBalancesToBalancesArray', () => {
    it('converts data correctly', () => {
        expect(convertStmBalancesToBalancesArray([stmBalance])).toMatchSnapshot();
    });
    it('converts undefined to empty array', () => {
        expect(convertStmBalancesToBalancesArray()).toEqual([]);
    });
});
