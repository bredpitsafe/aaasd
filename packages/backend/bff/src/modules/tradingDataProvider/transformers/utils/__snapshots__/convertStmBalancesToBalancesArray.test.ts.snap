// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`convertStmBalancesToBalancesArray > converts data correctly 1`] = `
[
  {
    "balances": [
      {
        "amount": 12.345,
        "assetId": 123,
        "assetName": "ASSET_NAME_1",
      },
      {
        "amount": 456.789,
        "assetId": 456,
        "assetName": "ASSET_NAME_2",
      },
      {
        "amount": 789.012,
        "assetId": 789,
        "assetName": "ASSET_NAME_3",
      },
    ],
    "instrumentId": 456,
    "instrumentName": "INSTRUMENT_ID",
    "platformTime": "2024-02-26 01:59:47.376",
    "robotId": 789,
    "robotName": "ROBOT_NAME",
    "virtualAccountId": 123,
    "virtualAccountName": "VIRTUAL_ACCOUNT",
  },
]
`;
